.justfile

```diff
@@ -0,0 +1,174 @@
+## Helper to print a message when calling `just`
+######################################################################
+[private]
+@default:
+  echo "Usage: just <recipe>"
+  just list-repo-recipes
+  echo "For further information, run 'just --help'"
+
+# List recipes in this file and from the calling directory
+[no-cd]
+[private]
+@usage:
+  echo "Usage: just <recipe>"
+  just list-local-recipes
+  just list-repo-recipes
+  echo "For further information, run 'just --help'"
+
+[no-cd]
+[private]
+@list-local-recipes:
+  echo '\nAvailable recipes:'
+  just --list --unsorted --list-heading ''
+
+[private]
+@list-repo-recipes:
+  echo "\nRepository recipes:"
+  just --list --unsorted --list-heading ''
+
+
+######################################################################
+## Helper to run a global command
+######################################################################
+
+# Runs yarn in the repository root
+[private]
+yarn *args:
+  yarn {{args}}
+
+
+######################################################################
+## Helper to run a command on an environmental condition
+######################################################################
+
+# Runs the provided command if `PROFILE` starts with `"dev"`
+[private]
+[no-cd]
+in-dev +command:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [[ {{ profile }} =~ dev.* ]]; then
+    echo "{{command}}" >&2
+    {{command}}
+  fi
+
+# Runs the provided command if in a pull request
+[private]
+[no-cd]
+in-pr +command:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [[ {{ github-event-name }} = pull_request ]]; then
+    echo "{{command}}" >&2
+    {{command}}
+  fi
+
+# Runs the provided command if not in a pull request
+[private]
+[no-cd]
+not-in-pr +command:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [[ {{ github-event-name }} != pull_request ]]; then
+    echo "{{command}}" >&2
+    {{command}}
+  fi
+
+
+######################################################################
+## Install scripts
+######################################################################
+
+[private]
+@install-cargo-hack:
+  cargo install cargo-hack --version 0.5.26 --locked
+
+[private]
+@install-cargo-nextest:
+  cargo install cargo-nextest --version 0.9.37 --locked
+
+[private]
+@install-cargo-script:
+  cargo install cargo-script --version 0.2.8 --locked
+
+[private]
+@install-llvm-cov:
+  cargo install cargo-llvm-cov --version 0.5.9 --locked
+
+[private]
+@install-cargo-insta:
+  cargo install cargo-insta --version 1.18.2 --locked
+
+
+######################################################################
+## Predefined commands
+######################################################################
+
+# Updates the compiler warnings in the `.cargo/config.toml` file.
+[private]
+[no-cd]
+lint-toml mode:
+  @cargo script "{{repo}}/.github/scripts/rust/lint.rs" {{mode}}
+
+# Runs all linting commands and fails if the CI would fail
+[no-cd]
+lint:
+  @just format -- --check
```

Comment:
This looks different to https://github.com/hashintel/hash/blob/b81c50a5c35fd9fcf73b5ef0ecc9994bd580f6bf/.github/workflows/rust.yml#L103

Does the `@just` syntax require `--`?

Reply:
Thanks for the information :)